src/core/public/ui_new_platform.test.mocks.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { coreMock } from './mocks';
+
+export const coreSetupMock = coreMock.createSetup();
+export const coreStartMock = coreMock.createStart();
+
+jest.doMock('ui/new_platform', () => {
+  return {
+    npStart: { core: coreStartMock },
+    npSetup: { core: coreSetupMock },
+  };
+});
```

Comment:
I think this might be easier to use if we move this to `src/legacy/ui/public/__mocks__/new_platform.ts` and allow people to get the mock by calling `jest.mock('ui/new_platform')`

Reply:
This makes it harder to access the mock, as you have to import from `ui/new_platform` and cast the type to `jest.Mocked<...>` but in the majority of cases tests just need the mock to exist and don't need to inspect it.